tests/CMakeLists.txt

```diff
@@ -152,10 +152,10 @@ function(pybind11_enable_warnings target_name)
     endif()
   endif()
 
-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND PYTHON_VERSION VERSION_LESS 3.0)
-    if(CMAKE_CXX_STANDARD STREQUAL "14")
+  if(CMAKE_CXX_STANDARD AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND PYTHON_VERSION VERSION_LESS 3.0)
```

Comment:
What happens if `CMAKE_CXX_STANDARD` is not defined?

Reply:
Then it is empty and this does not get added. I'm not being helpful here if `CMAKE_CXX_STANDARD` is not used. In the near future, the C++ handling will be improved and this likely will become a generator expression.